### PR TITLE
Cacheable: Fix type of dbid; Use override for serialize; Fix spelling

### PR DESCRIPTION
**Pull request by cb-mega**

### CODE REVIEW

File: include/mega/types.h [hunk 1]
```diff
@@ -248,16 +248,15 @@ const int FOLDERNODEKEYLENGTH = 16;
 typedef list<class Sync*> sync_list;
 
 // persistent resource cache storage
-struct Cachable
+class Cacheable
```
Comment (mattw-mega) [inline include/mega/types.h]: all members are public so struct is more natural than class - why change it?  Save a keyword ;-)

File: include/mega/types.h [hunk 2]
```diff
@@ -248,16 +248,15 @@ const int FOLDERNODEKEYLENGTH = 16;
...
 {
-    virtual bool serialize(string*) = 0;
-
-    int32_t dbid;
+public:
+    virtual ~Cacheable() = default;
```
Comment (mattw-mega) [inline include/mega/types.h]: the destructor is an implementation detail, why not leave it at the bottom?  serialize() is the most important member

File: include/mega/types.h (hunk 1 above)
Comment (cb-mega) [reply include/mega/types.h]: A struct shouldn't have virtual functions IMO.

File: include/mega/types.h (hunk 2 above)
Comment (cb-mega) [reply include/mega/types.h]: The destructor is a very important impl. detail for base classes. You need to know if it's virtual or not to know how you can use the base class.

**Pull request merged**